cornflow-server/cornflow/commands/permissions.py

```diff
@@ -163,13 +158,18 @@ def get_permissions_in_code_as_tuples(
     """
     Get the permissions in code as tuples.
     """
+    from cornflow.shared.const import ALL_DEFAULT_ROLES, GET_ACTION
+
     # Create base permissions using a set to avoid duplicates
     permissions_tuples = set()
 
     # Add permissions from ROLES_WITH_ACCESS
     for role, action in base_permissions_assignation:
         for view in resources_to_register:
             if role in view["resource"].ROLES_WITH_ACCESS:
+                # For custom roles (not in ALL_DEFAULT_ROLES), only grant GET access
+                if role not in ALL_DEFAULT_ROLES and action != GET_ACTION:
+                    continue
                 permissions_tuples.add((role, action, views_in_db[view["endpoint"]]))
```

Comment:
Is there a reason why you have this `if` here instead of having this in get_base_permission():
```
base_permissions_assignation = BASE_PERMISSION_ASSIGNATION + [
        (custom_role, GET_ACTION)
        for custom_role in all_custom_roles_in_access
]
```
instead of this:
```
base_permissions_assignation = BASE_PERMISSION_ASSIGNATION + [
        (custom_role, action)
        for custom_role in all_custom_roles_in_access
        for action in ALL_DEFAULT_ACTIONS
]
```
?